src/bmffimage.cpp

```diff
@@ -86,8 +87,9 @@ BmffImage::BmffImage(BasicIo::UniquePtr io, bool /* create */, size_t max_box_de
 }  // BmffImage::BmffImage
 
 std::string BmffImage::toAscii(uint32_t n) {
-  const auto p = reinterpret_cast<const char*>(&n);
-  std::string result(p, p + 4);
+  std::array<char, sizeof(uint32_t)> p;
+  std::memcpy(p.data(), &n, sizeof(uint32_t));
+  std::string result(p.begin(), p.end());
```

Comment:
I think this function might be a cleaner as a loop that mask the bottom byte and then does `n >>= 8` on each iteration. That way, there's no need to think about endianness.

Reply:
I just tried this.

godbolt output ballooned from 39 lines to 287. The code looks incredibly inefficient with operator new and memcpy being called.

Reply:
Second attempt went from 39 to 49. Maybe that's good enough.